bundles/org.openhab.binding.linktap/src/main/java/org/openhab/binding/linktap/protocol/frames/DeviceCmdReq.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.linktap.protocol.frames;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * The {@link DeviceCmdReq} is a request targetted to a device.
+ *
+ * @provides App: Device ID
+ *
+ * @author David Goodyear - Initial contribution
+ */
+@NonNullByDefault
+public class DeviceCmdReq extends TLGatewayFrame {
+
+    public DeviceCmdReq() {
+    }
+
+    public DeviceCmdReq(final int command) {
+        this.command = command;
+    }
+
+    /**
+     * Defines the targetted device ID
+     */
+    @SerializedName("dev_id")
+    @Expose
+    public String deviceId = EMPTY_STRING;
+
+    public String isValid() {
```

Comment:
I need to stop for tonight due to early commute tomorrow - I have had a think about this - potentially I am considering making a validation result class, that it can return. This will allow multiple validation errors to potentially be reported, and is would still keep it simple. Likely i'll refactor this to getValidationErrors() or a name more appropriate for the functionality. What was the test class you refer to above, in case I'm missing a trick?

Reply:
Looks good now.

Reply:
I just pushed a few changes - sorry maybe I miss understood, a bit of the last comment.